drivers/Makefile.dep

```diff
@@ -317,6 +317,13 @@ ifneq (,$(filter lc709203f,$(USEMODULE)))
   FEATURES_REQUIRED += periph_gpio_irq
 endif
 
+ifneq (,$(filter lis2dh12_int,$(USEMODULE)))
```

Comment:
Please let's restructure the complete block of `lis2dh12` dependencies a bit:
```
ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
  FEATURES_REQUIRED += periph_gpio
  FEATURES_REQUIRED += periph_spi
else
  FEATURES_REQUIRED += periph_i2c
endif

ifneq (,$(filter lis2dh12_int,$(USEMODULE)))
  FEATURES_REQUIRED += periph_gpio_irq
endif

ifneq (,$(filter lis2dh12%,$(USEMODULE)))
  USEMODULE += lis2dh12
endif
```
This avoids to use nested dependendy definitions and pulls in the `lis2dh12` automatically if any other pseudomodule is used.

Reply:
But if someone is only doing `USEMODULE += lis2dh12_int`, which transport will be used? I2C or SPI?

Reply:
Hm, good question. OK in former version it was by default `lis2dh12_i2c`
```
ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
  FEATURES_REQUIRED += periph_gpio
  FEATURES_REQUIRED += periph_spi
else
  FEATURES_REQUIRED += periph_i2c
endif
```
Ok, then I suggest (BTW, it was my first suggested version before I edited the comment):
```
ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
  FEATURES_REQUIRED += periph_gpio
  FEATURES_REQUIRED += periph_spi
else
  FEATURES_REQUIRED += periph_i2c
endif

ifneq (,$(filter lis2dh12_int,$(USEMODULE)))
  FEATURES_REQUIRED += periph_gpio_irq
endif

ifneq (,$(filter lis2dh12%,$(USEMODULE)))
  USEMODULE += lis2dh12
endif```
```
That is the difference would be
```diff
+ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_gpio
+  FEATURES_REQUIRED += periph_spi
+endif
+
+ifneq (,$(filter lis2dh12_i2c,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_i2c
+endif
+
+ifneq (,$(filter lis2dh12_int,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_gpio_irq
+endif
+
 ifneq (,$(filter lis2dh12%,$(USEMODULE)))
-  ifneq (,$(filter lis2dh12_spi,$(USEMODULE)))
-    FEATURES_REQUIRED += periph_gpio
-    FEATURES_REQUIRED += periph_spi
-  else
-    FEATURES_REQUIRED += periph_i2c
-  endif
   USEMODULE += lis2dh12
 endif
```

Reply:
It is also less nested and needs no error handling.